src/components/LoginPage.tsx

```diff
@@ -1,28 +1,60 @@
 import * as React from "react";
-import CheckboxLabel from "./SelectionControls/CheckboxLabel";
-import EmailTextField from "./TextFields/EmailTextField";
-import PasswordTextField from "./TextFields/PasswordTextField";
+import AlreadyHaveAnAccount from "./Level1/Buttons/AlreadyHaveAnAccount";
+import DontHaveAnAccount from "./Level1/Buttons/DontHaveAnAccount";
+import SignInButton from "./Level1/Buttons/SignInButton";
+import SignUpButton from "./Level1/Buttons/SignUpButton";
 
-class LoginPage extends React.PureComponent {
-  constructor(props: any) {
-    super(props);
-  }
+import TermsAndConditionsDialog from "./Level1/Dialogs/TermsAndConditionsDialog";
+
+import ForgotPasswordLink from "./Level1/Links/ForgotPasswordLink";
+
+import RememberMeCheckbox from "./Level1/SelectionControls/RememberMeCheckbox";
 
+import DateOfBirthTextField from "./Level1/TextFields/DateOfBirthTextField";
+import EmailTextField from "./Level1/TextFields/EmailTextField";
+import LoginTextFields from "./Level1/TextFields/LoginTextFields";
+import NameTextField from "./Level1/TextFields/NameTextField";
+import PasswordTextField from "./Level1/TextFields/PasswordTextField";
+
+export default class LoginPage extends React.PureComponent {
   public render() {
     const LoginStyle = {
       justifyContent: "right"
     };
 
     return (
       <div className="LoginDetails" style={LoginStyle}>
-        <EmailTextField />
+        <LoginTextFields type="email" />
+        <div />
+        <LoginTextFields type="password" />
+        <div />
+        <LoginTextFields type="name" />
+        <div />
+        <LoginTextFields type="dateOfBirth" />
```

Comment:
From user's perspective, I'd say `<DateOfBirthTextField />` is simpler to use than `<LoginTextFields type="dateOfBirth" />`. Basically, the name of a component is the "type" of the component and thus I'd prefer to use a specific name over a generic name with `type` property.

If you don't mind, I'd like to suggest to revert the last commit and discuss offline how to design our own reusable components.

Moreover, what I suggested before is to remove code redundancy in **button** components, not in text field components. ;) I just thought it might be a simple exercise. But as discussed yesterday, we don't have to do it now. We will have better idea on our own reusable components once we finish defining all the components needed for the pages.

Reply:
I did keep all the previous components just in case but if you wish to revert sure!

Reply:
Actually, can we keep this commit since it still has all the previous components? I'll change them back as it was but keep my class for later use and do another commit